ashpy/metrics/sliced_wasserstein_metric.py

```diff
@@ -0,0 +1,238 @@
+            raise ValueError(
+                "Minimum resolution cannot" "be smaller than the resolution"
+            )
+
+        self.resolution = resolution
+        self.resolution_min = resolution_min
+        self.patches_per_image = patches_per_image
+        self.patch_size = patch_size
+        self.random_sampling_count = random_sampling_count
+        self.random_projection_dim = random_projection_dim
+        self.use_svd = use_svd
+
+        self.children_real_fake = [
+            (
+                SingleSWD(model_selection_operator, logdir, 2 ** i, "real"),
+                SingleSWD(model_selection_operator, logdir, 2 ** i, "fake"),
+            )
+            for i in range(
+                int(np.log2(resolution)), int(np.log2(resolution_min)) - 1, -1
+            )
+        ]
+
+    def update_state(self, context: GANContext) -> None:
+        """
+        Update the internal state of the metric, using the information from the context object.
+
+        Args:
+            context (:py:class:`ashpy.contexts.GANContext`): An AshPy Context Object that carries
+                all the information the Metric needs.
+
+        """
+        updater = lambda value: lambda: self._metric.update_state(value)
+        for real_xy, noise in context.dataset:
+            real_x, real_y = real_xy
+
+            g_inputs = noise
+            if len(context.generator_model.inputs) == 2:
+                g_inputs = [noise, real_y]
+
+            fake = context.generator_model(
+                g_inputs, training=context.log_eval_mode == LogEvalMode.TRAIN
+            )
+
+            # check the resolution is the same as the one passed as input
+            resolution = real_x.shape[1]
+            if resolution != self.resolution:
+                raise ValueError(
+                    "Image resolution is not the same as the input resolution."
+                )
+
+            scores = sliced_wasserstein_distance(
+                real_x,
+                fake,
+                resolution_min=self.resolution_min,
+                patches_per_image=self.patches_per_image,
+                use_svd=self.use_svd,
+                patch_size=self.patch_size,
+                random_projection_dim=self.random_projection_dim,
+                random_sampling_count=self.random_sampling_count,
+            )
+
+            fake_scores = []
+
+            for i, couple in enumerate(scores):
+                self.children_real_fake[i][0].update_state(context, couple[0])
+                self.children_real_fake[i][1].update_state(context, couple[1])
+                fake_scores.append(tf.expand_dims(couple[1], axis=0))
+
+            fake_scores = tf.concat(fake_scores, axis=0)
+
+            self._distribute_strategy.experimental_run_v2(updater(fake_scores))
+
+    def model_selection(
+        self, checkpoint: tf.train.Checkpoint, global_step: tf.Variable
+    ) -> None:
+        """
+        Perform model selection for each sub-metric
```

Comment:
Since this is diferent from the father metric, ok to the documentation but add the parameters doc